wfl/fit/ref_error.py

```diff
@@ -1,213 +1,222 @@
 import warnings
-from pathlib import Path
-from pprint import pprint
-import json
+import re
 
 import numpy as np
 
 from wfl.configset import ConfigSet, OutputSpec
 from wfl.calculators.generic import run as generic_calc
 
 
-def calc(input_configs, output_configs, calculator,
-         ref_property_prefix, category_keys):
-    """Calculate errors for some potential relative to some reference data.
-    NOTE: several optional args such as properties and forces_by_components are not yet available.
+def calc(inputs, calc_property_prefix, ref_property_prefix,
+         config_properties=None, atom_properties=None, category_keys="config_type",
+         weight_property=None, calc_outputs=None, calc_autopara_info=None):
+    """calculate error for calculation results relative to stored reference values
 
     Parameters
     ----------
-    input_configs: ConfigSet
-        configs to check
-    output_configs: OutputSpec
-        optional location for calculated configs (otherwise will be in memory, and lost when function returns)
-    calculator: tuple(constructor, args, kwargs)
-        contructor for calculator with any args or kwargs
+    inputs: iterable(Atoms)
+        input configurations with reference and calculated values in Atoms.info/arrays
+    calc_property_prefix: str / Calculator / tuple(calculator_constructor, args, kwargs)
+        prefix to info/array keys for calculated properties, or calculator constructor
     ref_property_prefix: str
-        prefix to property names for reference values
-    category_keys: list(str)
-        list of info keys to use to create categories that configs are grouped into
+        prefix to info/array keys for reference properties
+    config_properties: list(str), default ["energy/atom", "virial/atom/comp"]
```

Comment:
Since the property is actually computed outside this function, can't you just add an `Atoms.info[f"{calc_property_prefix}_atomization_energy"]`, and then pass `"atomization_energy"` to `ref_error`?

Reply:
I guess my reasoning is that calculating atomization energy is more of a post-processing step, like calculating per-atom properties and it's not really needed anywhere else. But definitely, can pre-compute atomisation energies, e.g. in cli or just my own scripts (though it would be nice this to be done as seamlessly as possible).

Reply:
ok, I can add that together with the parity plots. Will leave this unresolved as a reminder.